apps/src/storage/dataBrowser/AddTableRow.jsx

```diff
@@ -29,13 +32,26 @@ class AddTableRow extends React.Component {
   }
 
   handleAdd = () => {
-    this.setState({isAdding: true});
-    FirebaseStorage.createRecord(
-      this.props.tableName,
-      _.mapValues(this.state.newInput, castValue),
-      () => this.setState(INITIAL_STATE),
-      msg => console.warn(msg)
-    );
+    try {
+      if (experiments.isEnabled(experiments.APPLAB_DATASETS)) {
+        this.props.hideError();
+      }
+      const record = _.mapValues(this.state.newInput, inputString =>
+        castValue(inputString, /* allowUnquotedStrings */ false)
+      );
+      this.setState({isAdding: true});
+      FirebaseStorage.createRecord(
+        this.props.tableName,
+        record,
+        () => this.setState(INITIAL_STATE),
+        msg => console.warn(msg)
+      );
+    } catch (e) {
```

Comment:
Are there any other possible errors that could be caught here?

Reply:
Like what?

Reply:
Nevermind. Looking at the previous version, there was no error handling. This makes the same set of assumptions. Good to go. 👍